keps/sig-storage/1682-csi-driver-skip-permission/README.md

```diff
@@ -0,0 +1,227 @@
+
+### Non-Goals
+
+ - This does not attempt to update existing CSIDrivers themselves.
+
+## Proposal
+
+We propose that the `CSIDriver` type include a field that defines if the volume 
+provided by the driver supports changing volume ownership. This will be enabled
+with a new feature gate, `CSIVolumeSupportFSGroup`.   
+
+### Risks and Mitigations
+
+- CSIDrivers will need to be redeployed after this field is introduced if the desired behavior is modified.
+- If a cluster enables the `CSIVolumeSupportFSGroup` feature gate and then this feature gate is disabled,
+such as due to an upgrade or downgrade, then the cluster will revert to the current behavior of examining
+volumes and attempting to apply volume ownerships and permissions based on the defined `fsGroup`.
+
+## Design Details
+
+Currently Volume permission and ownership change is done automatically for every volume. 
+As part of this proposal we will change the algorithm that modifies volume ownership and permissions
+for CSIDrivers to check the new field, and skip volume ownership modifications if it is found to be
+`false`.
+
+When defining a `CSIDriver`, we propose that `CSIDriver.Spec` be expanded to include a new field entitled 
+`SupportsFSGroup` which can have following possible values:
+
+ - `OnlyRWO` --> Current behavior. Attempt to modify the volume ownership and permissions to the defined `fsGroup` when the volume is 
+ mounted if accessModes is RWO.
+ - `Never` --> New behavior. Attach the volume without attempting to modify volume ownership or permissions.
+ - `Always` --> New behavior. Always attempt to apply the defined fsGroup to modify volume ownership and permissions.
+
+```go
+type CSIDriverFSGroupChangePolicy string
+
+const(
+    OnlyRWO CSIDriverFSGroupChangePolicy = "OnlyRWO"
+    Always CSIDriverFSGroupChangePolicy = "Always"
+    Never CSIDriverFSGroupChangePolicy = "Never"
+)
+
+type CSIDriverSpec struct {
+    // SupportsFSGroup ← new field
+    // Defines if the underlying volume supports changing ownership and 
+    // permission of the volume before being mounted. 
+    // If set to Always, SupportsFSGroup indicates that 
+    // the underlying volume supports volume ownership and permission
+    // changes, and the filesystem will always be modified to match the defined
+    // fsGroup.
+    // If set to Never, then the volume will be mounted without modifying
+    // the volume's ownership or permissions.
+    // Defaults to OnlyRWO, which results in the volume being examined
+    // and the volume ownership and permissions attempting to be updated
+    // if fsGroup is defined and accessModes is RWO.
+    // + optional
+    SupportsFSGroup *CSIDriverFSGroupChangePolicy
+}
+```
+### Test Plan
+
+A test plan will include the following tests:
+
+* Basic tests including a permutation of the following values:
+  - CSIDriver.Spec.SupportsFSGroup  (`nil`/`true`/`false`)
+  - PersistentVolumeClaimStatus.FSGroup (matching, non-matching)
```

Comment:
What is this field? Do you mean the fsGroupChangePolicy in the PodSecurityContext?

Reply:
Apologies - I was attempting to reference the AccessModes that currently control whether or not we apply the volume ownership/permissions. I thought it would be a good idea to include tests that if CSIDriver.Spec.SupportsFSGroup = OnlyRWO, then we kept the current behaviour of not performing modifications if the AccessMode isn't what we expect.